mojo/core/broker_castanets.cc

```diff
@@ -463,19 +578,74 @@ void BrokerCastanets::OnBufferSync2d(uint64_t guid_high,
           << ", width: " << width << ", stride: " << stride
           << ", fence_id: " << fence_id;
 
+  void* uncompressed_data = nullptr;
+  uint8_t* ptr = nullptr;
+
   scoped_refptr<base::CastanetsMemoryMapping> mapping =
       base::SharedMemoryTracker::GetInstance()->FindMappedMemory(guid);
+  if (!mapping) {
+    base::SharedMemoryCreateOptions options;
+    options.size = buffer_bytes;
+    base::subtle::PlatformSharedMemoryRegion handle =
+        base::CreateAnonymousSharedMemoryIfNeeded(guid, options);
+    CHECK(handle.IsValid());
+
+    void* memory = mmap(NULL, original_size + offset, PROT_READ | PROT_WRITE,
+                        MAP_SHARED, handle.GetPlatformHandle().fd, 0);
+    ptr = static_cast<uint8_t*>(memory);
+  }
```

Comment:
How about initializing |ptr| as |mapping->GetMemory()| even if |mapping| in null?
It seems to remove codes for |mapping| in line 607, 629.

Reply:
Yes, we can do that.. Updated in latest patchset.